packages/async-flow/test/async-flow-debug.test.js

```diff
@@ -0,0 +1,219 @@
+
+  const expectedRecording = {
+    length: recLen17,
+    0: error('p2 rejection:', '(Error#1)'),
+    1: error('Error#1:', reason.message),
+    2: consStack(2),
+    3: error('Error#1 ERROR_NOTE:', 'from host error', '(Error#2)'),
+    4: error(
+      'Error#1 ERROR_NOTE:',
+      'from flow',
+      'AsyncFlow1',
+      'defined at',
+      '(Error#3)',
+    ),
+    5: error(
+      'Error#1 ERROR_NOTE:',
+      'host rejection from call to (',
+      hOrch,
+      ').vow(...)',
+    ),
+    6: group('Nested 2 errors under Error#1'),
+    7: error('Error#2:', 'should be seen'),
+    8: consStack(8),
+    9: error('Error#2 ERROR_NOTE:', 'from guest error', '(Error#4)'),
+    [recLen17 - 7]: group('Nested error under Error#2'),
+    [recLen17 - 6]: error('Error#4:', 'should be seen'),
+    [recLen17 - 5]: consStack(recLen17 - 5),
+    [recLen17 - 4]: cons('groupEnd'),
+    [recLen17 - 3]: error('Error#3:', 'this stack'),
+    [recLen17 - 2]: consStack(recLen17 - 2),
+    [recLen17 - 1]: cons('groupEnd'),
+  };
+
+  // XXX: These "Sent as" messages when storing errors in the liveslots
+  // durable object state are noisy and spurious.
+  for (let i = 0; i < recLen17; i += 1) {
+    if (expectedRecording[i] === undefined) {
+      expectedRecording[i] = consSentAs(i);
+    }
+  }
+  t.like(recording, expectedRecording);
+  t.is(Object.keys(indexToStackRegex).length, 0, 'all stack entries tested');
+};
+
+/**
+ * @param {Zone} zone
+ * @param {number} [k]
+ */
+const prepareOrchestra = (zone, k = 1) =>
+  zone.exoClass(
+    'Orchestra',
+    undefined,
+    (factor, vow, resolver) => ({ factor, vow, resolver }),
+    {
+      scale(n) {
+        const { state } = this;
+        return k * state.factor * n;
+      },
+      vow() {
+        const { state } = this;
+        return state.vow;
+      },
+      resolve(x) {
+        const { state } = this;
+        state.resolver.resolve(x);
+      },
+      reject(x) {
+        const { state } = this;
+        state.resolver.reject(x);
+      },
+    },
+  );
+
+/** @typedef {ReturnType<ReturnType<prepareOrchestra>>} Orchestra */
+
+/**
+ * @param {ExecutionContext<unknown>} t
+ * @param {Zone} zone
+ */
+const testRejectionPlay = async (t, zone) => {
+  t.log('rejectionPlay started');
+  const vowTools = prepareVowTools(zone);
+  const { asyncFlow, adminAsyncFlow } = prepareTestAsyncFlowTools(t, zone, {
+    vowTools,
+  });
+  const makeOrchestra = prepareOrchestra(zone);
+  const { makeVowKit } = vowTools;
+
+  const { vow: v2, resolver: r2 } = zone.makeOnce('v2', () => makeVowKit());
+  const hOrch7 = zone.makeOnce('hOrch7', () => makeOrchestra(7, v2, r2));
+
+  // purposely violate rule that guestMethod is closed.
+  const { promise: promiseStep, resolve: resolveStep } = makePromiseKit();
+
+  const { guestMethod } = {
+    async guestMethod(gOrch7) {
+      const p2 = gOrch7.vow();
+
+      await eventLoopIteration();
+
+      const caughtP = p2.catch(e => {
```

Comment:
I would like to sanity check that the promise doesn't fulfill.

```suggestion
      const caughtP = p2.then(
        v => t.fail(`vow fulfilled with value ${v}`),
        e => {
```